packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -87,7 +122,72 @@ class CodeSizeController {
       name: treemapNodeName,
       byteSize: treemapNodeSize,
       childrenMap: childrenMap,
-      showDiff: showDiff,
+    )..addAllChildren(treemapNodeChildren);
+  }
+
+  /// Builds a tree with [TreemapNode] from [treeJson] which represents
+  /// the hierarchical structure of the tree.
+  TreemapNode generateDiffTree(
+    Map<String, dynamic> treeJson, {
+    DiffTreeType diffTreeType = DiffTreeType.combined,
+  }) {
+    var treemapNodeName = treeJson['n'];
```

Comment:
a lot of this code is now duplicated. Use helpers to breakup the duplicated code. It looks like everything before the diff code can be put into a helper and everything after the diff-specific code can be put in a helper. Then those helpers can be called from both `generateTree` and `generateDiffTree`

Reply:
Done